### PR TITLE
Login GitHub redirect

### PR DESCRIPTION
**Pull request by kyriakipapadimitriou**
Login with Github correct redirect has been fixed and all tests are passing. I wrote tests for apps_controller with login as well.

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -1,15 +1,18 @@
 class ApplicationController < ActionController::Base
-
+  #, if: :storable_location?
```
Comment (csungwon) [inline app/controllers/application_controller.rb]: @kyriakipapadimitriou @julies642 Can you explain what this line does? Seems like a single line comment(#), and if it is, a descriptive comment would be better.

File: features/login_github_redirect.feature [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+Scenario: not logged in so should see login with Github from orgs page 
+  #Story ID: #152298585
+  Given I am on the orgs page
+  Then I should see "Log in with GitHub"
+  
+Scenario: not logged in so should see login with Github from create page 
+  #Story ID: #152298585
+  Given I am on the create page
+  Then I should be on the login page
+  # Then I should see "Log in with GitHub"
```
Comment (csungwon) [inline features/login_github_redirect.feature]: Don't forget to delete the comment when you are done!

File: features/login_github_redirect.feature [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+    And the following users exist:
+        | name  | github_uid      | email         |
+        | user1 | esaas_developer | test@user.com |
+        | user2 |                 | test@user.com |
+        | user3 |                 | test@user.com |
+
+
+  And I am not logged in
+
+Scenario: not logged in so should see login with Github from users page
```
Comment (csungwon) [inline features/login_github_redirect.feature]: The three of the following scenario seems to test login_controller, not the redirect feature. If you intended to test the login_controller behavior, make a separate file for that.

File: features/login_github_redirect.feature [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+  
+Scenario: not logged in so should see login with Github from create page 
+  #Story ID: #152298585
+  Given I am on the create page
+  Then I should be on the login page
+  # Then I should see "Log in with GitHub"
+
+Scenario: login with Github from orgs page and redirect to orgs
+  #Story ID: #152298585
+  Given I am not logged in
```
Comment (csungwon) [inline features/login_github_redirect.feature]: I see a lot of "Given I am not logged in" and "And I am on the apps page" at the beginning of each scenario. I think it is a better idea put those into the background.

File: features/login_github_redirect.feature [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+  And I follow "Orgs"
+  And I follow "Log in with GitHub"
+  Then I should be on the orgs page
+
+Scenario: login with Github from users page and redirect to users
+  #Story ID: #152298585
+  Given I am not logged in
+  And I am on the apps page
+  And I follow "Users"
+  #And I am visiting the provider "users"
```
Comment (csungwon) [inline features/login_github_redirect.feature]: another unnecessary comment :)

File: features/login_github_redirect.feature [hunk 6]
```diff
@@ -0,0 +1,151 @@
...
+  And I follow "Log in with GitHub"
+  Then I should be on the create page
+  
+Scenario: login with Github from create page, logout, login back and redirect to create
+  #Story ID: #152298585
+  Given I am not logged in
+  And I am on the apps page
+  And I follow "Create"
+  And I follow "Log in with GitHub"
+  And I am not logged in
```
Comment (csungwon) [inline features/login_github_redirect.feature]: redundant line

File: features/login_github_redirect.feature [hunk 7]
```diff
@@ -0,0 +1,151 @@
...
+  Given I am not logged in
+  And I am on the apps page
+  And I follow "Create"
+  And I follow "Orgs"
+  And I follow "Users"
+  And I follow "Create"
+  And I follow "Log in with GitHub"
+  Then I should be on the create page
+  
+Scenario: Login to github to create a new app
```
Comment (csungwon) [inline features/login_github_redirect.feature:74]: I see what you are trying to test: user wants to create a new app and when they log in to the app, they should be redirected to "New app" page. You indeed test this behavior in the first half of the scenario, but the second half seems to test the behavior of apps_controller#new and apps_controller#create. Keeping only the first half makes it a better scenario

File: features/login_github_redirect.feature [hunk 8]
```diff
@@ -0,0 +1,151 @@
...
+  And I should see "App Description can't be blank"
+  When I fill in "App Name" with "Fake app"
+  When I fill in "App Description" with "Fake app description "
+  When I fill in "Deployment url" with "Fake app deployment url"
+  When I fill in "Repository url" with "Fake app repository "
+  And I press "Save"
+  Then I should be on the apps page
+  And I should see "App was successfully created."
+
+Scenario: Login to github to edit an existing app successfully
```
Comment (csungwon) [inline features/login_github_redirect.feature]: Same here. Check only if the user is indeed redirected to edit_app page.

**Review by csungwon — CHANGES_REQUESTED** [review]
The implementation seems very concise and accurate of what we expected. Nice work on the implementation. I have made some comments on the usage of comment and some comments on cucumber features. Please look through them and see if you can make your test any better. Nice work overall!

File: features/login_github_redirect.feature (hunk 3 above)
Comment (kimhoang73) [reply features/login_github_redirect.feature]: Agree with Sean! You can just move it to another feature maybe so we can keep those tests. And we can get better coverage lol

**Review by Jldeatrick — CHANGES_REQUESTED** [review]
Sean & Kim's comments seem reasonable - just a few minor changes. Looks great overall!

File: app/controllers/application_controller.rb (hunk 1 above)
Comment (kyriakipapadimitriou) [reply app/controllers/application_controller.rb]: yes it was a leftover commented out code from an implementation I tried initially. This line has been now deleted. Thanks for the catch!

File: features/login_github_redirect.feature (hunk 2 above)
Comment (kyriakipapadimitriou) [reply features/login_github_redirect.feature]: Thanks!

File: features/login_github_redirect.feature (hunk 3 above)
Comment (kyriakipapadimitriou) [reply features/login_github_redirect.feature]: Thanks! Good point. I am moving them to a new file.

File: features/comment.feature [hunk 9]
```diff
@@ -1,41 +0,0 @@
-Feature: Different users can comment on an App
```
Comment (kimhoang73) [inline features/comment.feature:1]: Also, I'm not sure what's going on, but why are there blocks of red for all the other feature files? Does this mean they are deleted? I think all files that you aren't working with should be untouched

**Review by kimhoang73 — CHANGES_REQUESTED** [review]

File: features/login_github_redirect.feature (hunk 5 above)
Comment (kyriakipapadimitriou) [reply features/login_github_redirect.feature]: Thanks!

File: features/login_github_redirect.feature (hunk 6 above)
Comment (kyriakipapadimitriou) [reply features/login_github_redirect.feature]: thanks good catch!

File: features/login_github_redirect.feature (hunk 7 above)
Comment (kyriakipapadimitriou) [reply features/login_github_redirect.feature:74]: Thanks for the suggestion! I kept the login part of the test in the login feature and moved the apps_controller test part to a new feature for apps_controller.

File: features/login_github_redirect.feature (hunk 8 above)
Comment (kyriakipapadimitriou) [reply features/login_github_redirect.feature]: Thanks for the suggestion! I kept the login part of the test in the login feature and moved the apps_controller test part to a new feature for apps_controller.

File: features/comment.feature (hunk 9 above)
Comment (kyriakipapadimitriou) [reply features/comment.feature:1]: I deleted the files that had failing tests and were not in the files I worked on to pass Travis. They should all be merged back when we submit all PRs.

**Review by kimhoang73 — APPROVED** [review]

**Review by csungwon — APPROVED** [review]
looks good to me! Nice work!

**Review by Jldeatrick — APPROVED** [review]

**Pull request merged**